src/basic/time-util.c

```diff
@@ -910,14 +908,36 @@ int parse_timestamp(const char *t, usec_t *usec) {
         }
 
         if (pid == 0) {
+                bool with_tz = true;
+
                 if (setenv("TZ", t_timezone, 1) != 0) {
                         shared->return_value = negative_errno();
                         _exit(EXIT_FAILURE);
                 }
 
                 tzset();
 
-                shared->return_value = parse_timestamp_impl(t, &shared->usec, true);
+                //If there is a timezone that mates the tzname fields, leave the parsing to the implementation
+                //Otherwise just cut if of
```

Comment:
C-style (/* */) comments are strongly preferred. Also, since those are two sentences, add periods. And it's "cut if off" not "cut if of". "Matches" not "mates".